backstory/src/test/java/com/google/sps/perspective/PerspectiveStoryAnalysisManagerTest.java

```diff
@@ -0,0 +1,189 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.perspective;
+
+import static org.mockito.Mockito.*;
+
+import au.com.origma.perspectiveapi.v1alpha1.PerspectiveAPI;
+import au.com.origma.perspectiveapi.v1alpha1.models.AnalyzeCommentRequest;
+import au.com.origma.perspectiveapi.v1alpha1.models.AnalyzeCommentResponse;
+import au.com.origma.perspectiveapi.v1alpha1.models.AttributeScore;
+import au.com.origma.perspectiveapi.v1alpha1.models.AttributeType;
+import au.com.origma.perspectiveapi.v1alpha1.models.Score;
+import com.google.sps.perspective.PerspectiveStoryAnalysisManager;
+import com.google.sps.perspective.data.APINotAvailableException;
+import com.google.sps.perspective.data.MockPerspectiveAPIFactory;
+import com.google.sps.perspective.data.NoAppropriateStoryException;
+import com.google.sps.perspective.data.StoryDecision;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.junit.Assert;
+import org.mockito.ArgumentCaptor;
+import org.mockito.ArgumentMatchers;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+import org.mockito.stubbing.ValidableAnswer;
+
+
+/** Quality tests for PerspectiveStoryAnalysisManager */
+@RunWith(JUnit4.class)
+public final class PerspectiveStoryAnalysisManagerTest {
+
+  /** a String of default text to use in tests */
+  private static final String DEFAULT_TEXT = "foo";
+
+  /** a factory to produce instances of mock Perspective APIs for the tests */
+  private static MockPerspectiveAPIFactory factory;
+
+
+  @Before 
+  public void setUp() {
+    // create a default instance of PerspectiveAPIFactory to use in any test 
+    // which doesn't need specified values (will just return the default value)
+    Map<AttributeType, Float> noDesiredScores = new HashMap<AttributeType, Float>();
+    factory = new MockPerspectiveAPIFactory(noDesiredScores);
+  }
+
+  /**
+   * Creates a PerspectiveStoryAnalysisManager object with the no-args constructor
+   * to check it constructs a non-null object.
+   */
+  @Test
+  public void checkNoArgsConstructor() {
+    try {
+      PerspectiveStoryAnalysisManager manager = new PerspectiveStoryAnalysisManager();
+      Assert.assertNotNull(manager);
+    }
+    catch(APINotAvailableException exception) {
+      Assert.fail(exception.toString()); // should fail if this exception is thrown
+    }
+  }
```

Comment:
When running this test on Github, this will throw since the API key is not checked in. I'm glad you thought of this, but seems reasonable to remove this since its behavior will be different depending on whether you're running this test locally or on Github.

Reply:
That makes sense. Thank you!

Reply:
Done